src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -63,14 +63,6 @@ QgsMeshLayer::QgsMeshLayer( const QString &meshLayerPath,
   {
     flags |= QgsDataProvider::FlagLoadDefaultStyle;
   }
-  if ( mReadFlags & QgsMapLayer::FlagTrustLayerMetadata )
```

Comment:
Can you explain why these are removed?

Reply:
Because this code is not at the right place.

In the constructor the `mReadFlags` is empty, it is set in the `readXml` method.

In the constructor no layer metadata are provided. Layer metadata are found in the layer XML.

In the constructor no flags are provided, there are only layer options.